app.js

```diff
@@ -0,0 +1,187 @@
+                  <input class="toggle" type="checkbox" ${checked}/>
+                  <label class="label">${contents}</label>
+                  <button class="destroy"></button>
+                  </div>
+            <input class="edit" value=${contents} />
+          </li>
+        `)
+      ).join('');
+      
+    // todo list의 총 item 갯수 표시
+    const length = todoList.length;
+    $todoCount.innerHTML = `총 <strong>${length}</strong> 개`;
+}
+
+// 사용자가 입력창에서 enter를 누를 때 동작
+$todoInput.addEventListener('keyup', ({ target, key }) => {
+    const contents = target.value;
+
+    // 눌린 키가 엔터가 아니거나 입력된 내용이 공백밖에 없으면 리턴
+    if (key !== "Enter" || !contents.trim()) return;
+
+    const id = newGuid(); // id 부여 (하단 newGuid() 함수 참고)
+    const completed = '';
+    const checked = '';
+    
+    todoList.push({ id, contents, completed, checked}); // 배열에 목록 정보 추가
+    target.value = ''; // 입력창 초기화
+
+    todoListRender(); // 목록 render
+})
+
+// 체크박스 클릭시 동작하는 이벤트 핸들러
+$todoList.addEventListener('click', ({target}) => {
+    // 클릭된 요소가 체크박스가 아니면 리턴
+    if (target.className !== 'toggle') return;
+    
+    // 가장 가까운 li 부모요소 찾기
+    const $li = target.closest('li');
+    const { id } = $li.dataset; // li요소에서 id 정보 가져오기
+
+    const idx = todoList.findIndex((todo) => todo.id === id); // 배열에서 해당 id를 가진 목록정보의 인덱스 찾아 idx에 저장
+    const completed = todoList[idx].completed; 
+    const checked = todoList[idx].checked;
+    
+    todoList[idx].completed = completed ? '' : 'completed';
+    todoList[idx].checked = checked ? '' : 'checked';
+    $li.className = completed ? '' : 'completed';
+    
+})
+
+// x 버튼 클릭시 이벤트 핸들러
+$todoList.addEventListener('click', ({target}) => {
+    // 클릭된 요소가 x 버튼이 아니면 리턴
+    if(target.className !== 'destroy') return;
+
+    // 가장 가까운 li 부모요소 찾기
+    const $li = target.closest('li');
+    const { id } = $li.dataset;
+
+    const idx = todoList.findIndex((todo) => todo.id === id);
+
+    // confirm 확인창에서 취소누르면 리턴
+    if(!confirm('정말로 삭제하시겠습니까?')) return;
+
+    todoList.splice(idx, 1); // 해당 요소 삭제
+
+    todoListRender(); // todoList 다시 불러오기
+});
+
+// 더블클릭시 수정모드로 전환
+$todoList.addEventListener('dblclick', ({target}) => {
+    if(target.className !== 'label') return;
+    
+    // 이미 수정하려고 더블클릭한 요소가 있었다면 자동 해제시킴
+    cancelEditMode();
+
+    const { id } = target.dataset;
+
+    // 수정하는 화면으로 모습을 바꿈
+    let $li = target.closest('li');
+    $li.classList.toggle('editing');
+    $childLabel = $li.querySelector('.edit');
+    $childLabel.focus();
+});
+
+function cancelEditMode(){
+    $todoList.querySelectorAll('li').forEach($item =>{
+        if($item.classList.contains('editing')){
+            $item.classList.toggle('editing');
+        }
+    });
+}
+
+$todoList.addEventListener('keyup', ({target, key}) => {
+    if(key == 'Escape'){ // esc키를 눌렀을 때
+        cancelEditMode();
+        todoListRender(); // 로딩 안 하면 esc누르기 전에 작성된 내용이 더블클릭할시 다시 나타남
+        return;
+    }
+    if(key == 'Enter'){ // enter키를 눌렀을 때
```

Comment:
```suggestion
    if(key === 'Enter'){ // enter키를 눌렀을 때
```

다른 부분의 비교 연산자는 모두 일치(===), 불일치(!==) 연산자를 사용하신걸로 보아 아마 오타로 동등(==), 부등(!=) 연산자를 사용하신거 같긴 한데 혹시 오타가 아니라면 어떤 이유로 ```==```을 사용하신건지 알 수 있을까요?

Reply:
사실 큰 이유는 없고 이벤트 객체의 프로퍼티랑 비교하는 거라 혹시나(?)해서 `==`를 썼던 것 같네요. 지금보니 그냥 문자열인데 권장되는 `===`를 쓰도록 수정하겠습니다.
수정해보면서 localStorage 심화 기능도 같이 해볼 예정인데 말씀해주신 것 처럼 `JSON.stringfy()`과 `JSON.parse()`를 활용해 관리하도록 구현해봐야겠네요. 꼼꼼한 리뷰 정말 감사드립니다 👍